fuse_core/include/fuse_core/autodiff_local_parameterization.h

```diff
@@ -190,7 +190,7 @@ bool AutoDiffLocalParameterization<PlusFunctor, MinusFunctor, kGlobalSize, kLoca
   return ceres::internal::AutoDiff<PlusFunctor, double, kGlobalSize, kLocalSize>
     ::Differentiate(*plus_functor_, parameter_ptrs, kGlobalSize, x_plus_delta, jacobian_ptrs);
 #else
-  return ceres::internal::AutoDifferentiate<ceres::internal::StaticParameterDims<kGlobalSize, kLocalSize>>(
+  return ceres::internal::AutoDifferentiate<kGlobalSize, ceres::internal::StaticParameterDims<kGlobalSize, kLocalSize>>(
```

Comment:
The addition of the `kNumResiduals` template argument to `ceres::internal::AutoDifferentiate` looks like it was introduced in `2.1.0`, based on https://github.com/ceres-solver/ceres-solver/commit/e7a30359ee754057f9bd7b349c98c291138d91f4

Does this still work with `2.0.0`?

If not, we should add an `#elseif` for `2.0.0` with the current implementation, similar to the `#if !CERES_VERSION_AT_LEAST(2, 0, 0)` above, so the new implementation goes inside the `#else` block.

```suggestion
#elif !CERES_VERSION_AT_LEAST(2, 1, 0)
  return ceres::internal::AutoDifferentiate<ceres::internal::StaticParameterDims<kGlobalSize, kLocalSize>>(
#else
  return ceres::internal::AutoDifferentiate<kGlobalSize, ceres::internal::StaticParameterDims<kGlobalSize, kLocalSize>>(
```

I also want to keep this fork in sync with upstream as much as possible. Could you create two separate PRs, one for this Ceres solver API change in 2.1.0, and another one for the missed `geometry_msgs` depend, please?

Then I'd take care of sending them upstream as well, and get them merged into our fork while they're reviewed upstream.

Reply:
As discussed offline, if we click on the `...` button in https://github.com/ceres-solver/ceres-solver/commit/e7a30359ee754057f9bd7b349c98c291138d91f4, we can see the change is in: [2.1.0](https://github.com/ceres-solver/ceres-solver/releases/tag/2.1.0)  [2.1.0rc2](https://github.com/ceres-solver/ceres-solver/releases/tag/2.1.0rc2) [2.1.0rc1](https://github.com/ceres-solver/ceres-solver/releases/tag/2.1.0rc1) [2.0.0](https://github.com/ceres-solver/ceres-solver/releases/tag/2.0.0) [2.0.0rc2](https://github.com/ceres-solver/ceres-solver/releases/tag/2.0.0rc2) [2.0.0rc1](https://github.com/ceres-solver/ceres-solver/releases/tag/2.0.0rc1)

It's also been confirmed it builds in 2.0.0.